### DISCUSSION TITLE
Disable Starlette/FastAPI TraceBack

### DISCUSSION THREAD

**Discussion by Mte90** [zauberzeug/nicegui#4066] [Q&A]
### Question

If there is an error in the code the amount of stuff showed is a lot and find the real error in the console output is very difficult.
There is a way to get only the first part of the error in the console?

**Comment by falkoschindler** [comment]
Hi @Mte90,

This sounds like a rather general Python question. Since we are running on FastAPI and uvicorn, the stacktraces can get quite long.
There is `sys.tracebacklimit`, but it might not be exactly what you're looking for.

**Reply by Mte90** [reply]
mainly the issue is that the traceback about the error that I am looking on is at the top of the log item, like at Warning and the rest of the log is just starlette and whatever else involce but is very loooooong.

**Comment by simonryf** [comment]
If you are open to use Rich logging ([https://rich.readthedocs.io](https://rich.readthedocs.io))
you can use: `tracebacks_suppress` and add stuff like fastapi,uvicorn,starlette...
I don't know if there is something similar in the default logging for python.

**Reply by Mte90** [reply]
I wil ltake a look, because as nicegui is a wrapper around those technologies I was thinking that maybe there is a parameter in nicegui to disable the trraceback for those.

**Comment by Mte90** [comment]
* https://github.com/encode/starlette/issues/891
* https://github.com/fastapi/fastapi/issues/2019
* https://github.com/fastapi/typer/discussions/596

It is something that you have to define on starlette initialization about the logging or in rich initialization. I think that it is something to be handled by nicegui.

**Comment by falkoschindler** [comment]
NiceGUI allows to register custom exception handlers. So we can remove the default handler and add one that only prints the last 3 layers:
```py
app._exception_handlers.clear()
app.on_exception(lambda e: print(''.join(traceback.format_exception(e)[-3:]).strip()))
```

Demo:
```py
def notify():
    ui.notify(1/0)

def handle_click():
    notify()

ui.button('Click me', on_click=handle_click)
```

**Reply by Mte90** [reply]
Awesome!
It works, it is missing the `import traceback`. I suggest to adding it to the docs.
